src/fileprovider.cs

```diff
@@ -0,0 +1,301 @@
+		NSString ItemIdentifier { get; }
+
+		[Advice ("Use 'NSFileProviderItemIdentifierExtensions.GetValue (ParentItemIdentifier)' to get a 'NSFileProviderItemIdentifier' enum.")]
+		[Abstract]
+		[Export ("parentItemIdentifier")]
+		NSString ParentItemIdentifier { get; }
+
+		[Abstract]
+		[Export ("filename")]
+		string Filename { get; }
+
+		[Abstract]
+		[Export ("typeIdentifier")]
+		string TypeIdentifier { get; }
+
+		[Export ("capabilities")]
+		NSFileProviderItemCapabilities GetCapabilities ();
+
+		[return: NullAllowed]
+		[Export ("documentSize", ArgumentSemantic.Copy)]
+		NSNumber GetDocumentSize ();
+
+		[return: NullAllowed]
+		[Export ("childItemCount", ArgumentSemantic.Copy)]
+		NSNumber GetChildItemCount ();
+
+		[return: NullAllowed]
+		[Export ("creationDate", ArgumentSemantic.Copy)]
+		NSDate GetCreationDate ();
+
+		[return: NullAllowed]
+		[Export ("contentModificationDate", ArgumentSemantic.Copy)]
+		NSDate GetContentModificationDate ();
+
+		[return: NullAllowed]
+		[Export ("lastUsedDate", ArgumentSemantic.Copy)]
+		NSDate GetLastUsedDate ();
+
+		[return: NullAllowed]
+		[Export ("tagData", ArgumentSemantic.Copy)]
+		NSData GetTagData ();
+
+		[return: NullAllowed]
+		[Export ("favoriteRank", ArgumentSemantic.Copy)]
+		NSNumber GetFavoriteRank ();
+
+		[Export ("isTrashed")]
+		bool IsTrashed ();
+
+		[Export ("isUploaded")]
+		bool IsUploaded ();
+
+		[Export ("isUploading")]
+		bool IsUploading ();
+
+		[return: NullAllowed]
+		[Export ("uploadingError", ArgumentSemantic.Copy)]
+		NSError GetUploadingError ();
+
+		[Export ("isDownloaded")]
+		bool IsDownloaded ();
+
+		[Export ("isDownloading")]
+		bool IsDownloading ();
+
+		[return: NullAllowed]
+		[Export ("downloadingError", ArgumentSemantic.Copy)]
+		NSError GetDownloadingError ();
+
+		[Export ("isMostRecentVersionDownloaded")]
+		bool IsMostRecentVersionDownloaded ();
+
+		[Export ("isShared")]
+		bool IsShared ();
+
+		[Export ("isSharedByCurrentUser")]
+		bool IsSharedByCurrentUser ();
+
+		[return: NullAllowed]
+		[Export ("ownerNameComponents")]
+		NSPersonNameComponents GetOwnerNameComponents ();
+
+		[return: NullAllowed]
+		[Export ("mostRecentEditorNameComponents")]
+		NSPersonNameComponents GetMostRecentEditorNameComponents ();
+
+		[return: NullAllowed]
+		[Export ("versionIdentifier")]
+		NSData GetVersionIdentifier ();
+
+		[return: NullAllowed]
+		[Export ("userInfo")]
+		NSDictionary GetUserInfo ();
+	}
+
+	[iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSFileProviderManager {
+
+		[Static]
+		[Export ("defaultManager", ArgumentSemantic.Strong)]
+		NSFileProviderManager DefaultManager { get; }
+
+		[Export ("signalEnumeratorForContainerItemIdentifier:completionHandler:")]
+		// Not Async'ified on pruppose, because this can switch from app to extension.
+		void SignalEnumerator (/*[BindAs (typeof (NSFileProviderItemIdentifier))]*/ NSString containerItemIdentifier, Action<NSError> completion);
```

Comment:
let's not merge until it's ready
and, if it pass intro, let's update the PR with what we intend to ship